### PR TITLE
add segment.io into satellite GUI

### PR DESCRIPTION
**Pull request by VinozzZ**
What: 
Create a vue-plugin for segment.io analytics
Why:
Marketing team would like to be able to track user activities on the satellite GUI

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

**Comment by stefanbenten** [comment]
Is this an Opt-In or Opt-Out Functionality?

**Comment by VinozzZ** [comment]
> Is this an Opt-In or Opt-Out Functionality?

From my understanding, this is a Opt-Out functionality.

File: web/satellite/src/utils/doNotTrack.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+  
+    if (
+      doNotTrackOption.charAt(0)  === '1' ||
+      doNotTrackOption === 'yes'
+    ) {
+      return true;
+    }
+  
+    return false;
```
[the remark this answers is not in the document]
Comment (VinozzZ) [reply web/satellite/src/utils/doNotTrack.ts]: good point. I changed it

File: web/satellite/src/views/forgotPassword/ForgotPassword.vue [hunk 2]
```diff
@@ -46,10 +47,12 @@
                     this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Please look for instructions at your email');
                 },
                 onBackToLoginClick: function() {
+                    this.$segment.track(EVENTS.CLICKED_BACK_TO_LOGIN);
                     this.$router.push(ROUTES.LOGIN.path);
                 },
                 onLogoClick: function () {
                    location.reload();
+                   this.$segment.track(EVENTS.CLICKED_LOGO);
```
Comment (rikysya) [inline web/satellite/src/views/forgotPassword/ForgotPassword.vue]: may be it is better to track the event before page is reloaded? If it works than its fine. It just that it is more clear that way in my opinion.

Comment (VinozzZ) [reply web/satellite/src/views/forgotPassword/ForgotPassword.vue]: That's a good point

**Pull request merged**